symbol-overlay.el

```diff
@@ -259,23 +276,26 @@ depending on SCOPE and WINDOW."
   "Highlight symbol at point when there are more than 2 occurrences.
 This only effects symbols in the current displayed window."
   (when symbol-overlay-mode
-    (let ((case-fold-search nil)
-	  (symbol (symbol-overlay-get-symbol nil t))
-	  p)
-      (when (and symbol (not (symbol-overlay-assoc symbol)))
-	(symbol-overlay-remove-temp)
-	(save-excursion
-	  (save-restriction
-	    (symbol-overlay-narrow symbol-overlay-scope t)
-	    (goto-char (point-min))
-	    (re-search-forward symbol nil t)
-	    (save-match-data
-	      (while (re-search-forward symbol nil t)
-		(symbol-overlay-put-one symbol)
-		(or p (setq p t))))
-	    (when p
-	      (symbol-overlay-put-one symbol)
-	      (setq symbol-overlay-temp-symbol symbol))))))))
+    (let* ((case-fold-search nil)
+           (symbol (symbol-overlay-get-symbol nil t))
+           (should-put-overlay (and symbol
```

Comment:
apart from adding an explicit `should-put-overlay` with a `ignored-p` check this is just reindentation.